ethereum-explorer/src/map_contract_events.rs

```diff
@@ -0,0 +1,73 @@
+use crate::pb::eth::event::v1::Events;
+use crate::pb::eth::event::v1::Event;
+use substreams_ethereum::pb::eth::v2::Block;
+use substreams::Hex;
+use substreams_ethereum::pb::eth::v2::Log;
+use substreams_ethereum::pb::eth::v2::TransactionTrace;
+use substreams::errors::Error;
+use crate::util;
+
+#[substreams::handlers::map]
+fn map_contract_events(contract_address: String, blk: Block) -> Result<Events, Error> {
```

Comment:
Ok, the way it's implemented today is actually missing a lot of logs. This is because the filtering for the `contract_address` is done actually at the `transaction` level. However, transactions contains internal call (a.k.a internal transactions) that can also emit logs.

The way the code is structured now will miss events perform as part of the execution of the transaction. I think it will be good with sit together Monday to go through those quirks of the model and also show you what we have today in `substreams-ethereum` as helpers to make the tasks that you want to perform easier to implement.